test/gator/validate/test.bats

```diff
@@ -0,0 +1,204 @@
+  if [ "$?" -ne 0 ]; then
+    printf "ERROR: got exit status %s but wanted 0\n" "$?"
+    exit 1
+  fi
+}
+
+@test "correctly finds violations from objects in a filesystem" {
+  ! bin/gator validate --filename="$BATS_TEST_DIRNAME/fixtures/policies/default" --filename="$BATS_TEST_DIRNAME/fixtures/manifests/no-policies/with-violations.yaml"
+}
+
+@test "expects user to input data" {
+  run bin/gator validate
+  [ "$status" -eq 1 ]
+  err_substring="no input data identified"
+  match_substring "${output[*]}" "${err_substring}"
+}
+
+@test "disallows invalid template" {
+  run bin/gator validate --filename="$BATS_TEST_DIRNAME/fixtures/manifests/invalid-resources/template.yaml"
+  [ "$status" -eq 1 ]
+  err_substring="reading yaml source"
+  match_substring "${output[*]}" "${err_substring}"
+}
+
+@test "disallows invalid constraint" {
+  run bin/gator validate --filename="$BATS_TEST_DIRNAME/fixtures/manifests/invalid-resources/constraint.yaml"
+  [ "$status" -eq 1 ]
+  err_substring="reading yaml source"
+  match_substring "${output[*]}" "${err_substring}"
+}
+
+@test "outputs valid json when flag is specified" {
+  run bin/gator validate --filename="$BATS_TEST_DIRNAME/fixtures/manifests/with-policies/with-violations.yaml" --output=json
+  [ "$status" -eq 1 ]
+  # uses jq version `jq-1.6`
+  if ! (echo -n "${output[*]}" | jq); then
+    printf "ERROR: expected output to be valid json\n"
+    printf "OUTPUT: %s\n" "${output[*]}"
+    exit 1
+  fi
+}
+
+@test "outputs valid yaml when flag is specified" {
+  run bin/gator validate --filename="$BATS_TEST_DIRNAME/fixtures/manifests/with-policies/with-violations.yaml" -o=yaml
+  [ "$status" -eq 1 ]
+  # Confirm we still get our violation output
+  want_msg="Container <tomcat> in your <Pod> <test-pod1> has no <readinessProbe>" 
+  match_yaml_msg "${output[*]}" "${want_msg}"
+}
+
+@test "correctly ingests files of different extensions, skipping bad extensions, and producing correct violations" {
+  run bin/gator validate --filename="$BATS_TEST_DIRNAME/fixtures/manifests/different-extensions" -o=yaml
+  [ "$status" -eq 1 ]
+
+  # Confirm we still get our violation output
+  want_msg="Container <tomcat> in your <Pod> <test-pod1> has no <readinessProbe>" 
+  match_yaml_msg "${output[*]}" "${want_msg}"
+}
+
+@test "enforcementAction=deny causes 1 exit status and violations output" {
+  run bin/gator validate \
+    -f="$BATS_TEST_DIRNAME/fixtures/policies/default/template_k8srequiredprobes.yaml" \
+    -f="$BATS_TEST_DIRNAME/fixtures/policies/enforcement_action/k8srequiredprobes/deny.yaml" \
+    -f="$BATS_TEST_DIRNAME/fixtures/manifests/no-policies/with-violations.yaml" \
+    -o=yaml
+
+  [ "$status" -eq 1 ]
+
+  # Confirm we still get our violation output
+  want_msg="Container <tomcat> in your <Pod> <test-pod1> has no <readinessProbe>" 
+  match_yaml_msg "${output[*]}" "${want_msg}"
+}
+
+@test "enforcementAction=warn causes 0 exit status and violations output" {
```

Comment:
We can lump these in with enforcementAction=<anythingelse> test below.

These are e2e tests, as such we're mainly testing plumbing. Detailed behavior can be tested more efficiently via unit tests.

Generally, a large # of e2e tests should be discouraged, since bash is much more brittle to write in than golang.

Reply:
extra tests removed